### ISSUE TITLE
ARM64 R2R: Assertion failed '!"Incompatible types for gtNewTempAssign"'

### ISSUE THREAD

**Issue by kyulee1** [dotnet/coreclr#5642]
JIT\SIMD\VectorReturn_ro\VectorReturn_ro.cmd
```
BEGIN EXECUTION
" C:\Users\ddjoin\scratch\crossgen.exe" /Platform_Assemblies_Paths C:\Users\ddjoin\scratch;C:\Users\ddjoin\CoreClrTests-afa7fe5486fc90fec7df52bd134b789c568c8b45\JIT\SIMD\VectorReturn_ro\ VectorReturn_ro.exe
               [000235] --CXG+-------             *  call r2r_ind struct System.Numerics.Vector2.op_Addition
               [000409] -------------             |  /--*      void  
               [000408] -------------             |  |  \--*  lclFld    float  V11 tmp10        [+4]
                                                  |  |  \--*    float  V11.X (offs=0x00) -> V22 tmp21        
                                                  |  |  \--*    float  V11.Y (offs=0x04) -> V23 tmp22        
               [000411] ------------- arg0 d0,d1  +--*      void  
               [000410] -------------             |  \--*  lclFld    float  V11 tmp10        [+0]
                                                  |  \--*    float  V11.X (offs=0x00) -> V22 tmp21        
                                                  |  \--*    float  V11.Y (offs=0x04) -> V23 tmp22        
               [000413] -------------             |  /--*      void  
               [000412] -------------             |  |  \--*  lclFld    float  V10 tmp9         [+4]
                                                  |  |  \--*    float  V10.X (offs=0x00) -> V20 tmp19        
                                                  |  |  \--*    float  V10.Y (offs=0x04) -> V21 tmp20        
               [000415] ------------- arg1 d2,d3  \--*      void  
               [000414] -------------                \--*  lclFld    float  V10 tmp9         [+0]
                                                     \--*    float  V10.X (offs=0x00) -> V20 tmp19        
                                                     \--*    float  V10.Y (offs=0x04) -> V21 tmp20        
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.22220.0
Copyright (c) Microsoft Corporation.  All rights reserved.


Assert failure(PID 416 [0x000001a0], Thread: 17088 [0x42c0]): Assertion failed '!"Incompatible types for gtNewTempAssign"' in 'VectorTest:F2(float):struct' (IL size 97)

    File: d:\j\workspace\arm64_cross_c---67295489\src\jit\gentree.cpp Line: 11489
    Image: C:\Users\ddjoin\scratch\crossgen.exe

Crossgen failed with exitcode - 123456789
```

**Comment by kyulee1** [comment]
This is basically a multi-reg return case with HFA type (two floats for this case), which is not yet implemented. The current implementation handles as if two floats are fit into one integer register (long) at importer, and later on we got type mismatch assertion like this -- struct vs. long. We should handle this struct type as it is using two registers, s0/s1.

So reassign it to Brian.

**Comment by briansull** [comment]
I am testing a fix for this

**Comment by kyulee1** [comment]
Fixed via #6035
